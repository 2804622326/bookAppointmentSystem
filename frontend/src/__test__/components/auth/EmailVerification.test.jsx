import React from 'react';
import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import EmailVerification from '../../../components/auth/EmailVerification.jsx';

jest.mock('../../../components/auth/AuthService.js', () => ({
  verifyEmail: jest.fn(() => Promise.resolve({ message: 'VALID' })),
  resendVerificationToken: jest.fn(() => Promise.resolve({ message: 'resent' })),
}));

jest.mock('../../../components/common/ProcessSpinner.jsx', () => () => <div data-testid='spinner' />);

beforeEach(() => {
  window.history.pushState({}, '', '/verify?token=test');
});

test('shows verification success and snapshot', async () => {
  const { asFragment } = render(<EmailVerification />);
  const { verifyEmail } = require('../../../components/auth/AuthService.js');
  await waitFor(() => expect(verifyEmail).toHaveBeenCalledWith('test'));
  expect(screen.getByText(/successfully verified/i)).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});

test('resends token when link expired', async () => {
  const { verifyEmail, resendVerificationToken } = require('../../../components/auth/AuthService.js');
  verifyEmail.mockRejectedValueOnce({ response: { data: { message: 'EXPIRED' } } });
  const { asFragment } = render(<EmailVerification />);
  await waitFor(() => screen.getByRole('button', { name: /resend verification link/i }));
  fireEvent.click(screen.getByRole('button', { name: /resend verification link/i }));
  await waitFor(() => expect(resendVerificationToken).toHaveBeenCalled());
  expect(asFragment()).toMatchSnapshot();
});
