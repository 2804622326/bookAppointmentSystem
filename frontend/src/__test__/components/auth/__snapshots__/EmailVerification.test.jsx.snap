// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resends token when link expired 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-center  mt-lg-5"
  >
    <div
      class="col-12 col-md-6"
    >
      <div
        class="alert alert-success"
        role="alert"
      >
        resent
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`shows verification success and snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-center  mt-lg-5"
  >
    <div
      class="col-12 col-md-6"
    >
      <div
        class="alert alert-success"
        role="alert"
      >
        Your email has been successfully verified, you can proceed to login.
      </div>
    </div>
  </div>
</DocumentFragment>
`;
